### ISSUE TITLE
[Bug?]: workspaces foreach with --parallel flag don't work on GitHub Actions

### ISSUE THREAD

**Issue by FezVrasta** [yarnpkg/berry#3512]
### Self-service

- [ ] I'd be willing to implement a fix

### Describe the bug

Running two or more commands in parallel using `yarn workspaces foreach -piv --from 'pkg_{a,b}' run serve` doesn't work on GitHub Actions, only the first command is executed.

### To reproduce

I prepared a [repro repository here](https://github.com/FezVrasta/yarn-foreach-gha-error-repro), and you can observe a CI run where the logs show only the first command has been executed.

https://github.com/FezVrasta/yarn-foreach-gha-error-repro/runs/3756167117?check_suite_focus=true

### Environment

```shell
GitHub Actions machines
```

### Additional context

The same command works correctly on my local macOS machine. I tried to execute some "build" commands (that ran webpack and tsc) and I could observe how the `--parallel` flag seems to be ignored and they are actually run in series.

I think the issue may be that GitHub Actions machines have 2 cores, so Yarn divides by 2 and gets a single job.

A possible solution could be to either warn the user that the `--parallel` flag is being ignored because only one job is getting executed at a time. Or tweak the algorithm to take in account CI environments and use all the available cores.

**Comment by yarnbot** [comment]
Hi! 👋

This issue looks stale, and doesn't feature the `reproducible` label - which implies that you didn't provide a working reproduction using [Sherlock](https://yarnpkg.com/advanced/sherlock). As a result, it'll be closed in a few days unless a maintainer explicitly vouches for it or you edit your first post to include a formal reproduction (you can use the [playground](http://yarnpkg.com/playground) for that).

Note that we require Sherlock reproductions for long-lived issues (rather than standalone git repositories or similar) because we're a small team. Sherlock gives us the ability to check which bugs are still affecting the master branch at any given point, and decreases the amount of code we need to run on our own machines (thus leading to faster bug resolutions). It helps us help you! 😃

If you absolutely *cannot* reproduce a bug on Sherlock (for example because it's a Windows-only issue), a maintainer will have to manually add the `upholded` label. Thanks for helping us triaging our repository! 🌟

**Comment by FezVrasta** [comment]
Bump

**Comment by Leksat** [comment]
Looks like a duplicate of https://github.com/yarnpkg/berry/issues/4147
